libipt/src/pt_block_decoder.c

```diff
@@ -2431,12 +2435,14 @@ pt_blk_proceed_no_event_fill_cache(struct pt_block_decoder *decoder,
 	if (status < 0)
 		return status;
 
+	
+	int valid_cache = pt_bce_is_valid(bce);
```

Comment:
Please declare variables at the beginning of the function.  We also need to find a better name, because valid_cache refers to an older cache entry that might have been updated when we use the variable.  E.g. keep_filling, or just fill.